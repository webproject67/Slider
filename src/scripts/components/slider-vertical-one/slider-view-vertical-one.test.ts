import SliderModel from '../model/slider-model';
import SliderViewVerticalOne from './slider-view-vertical-one';

const main = document.createElement('div');
const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  range: 'one',
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'vertical'
};

const model = new SliderModel(main, state);
const slider = new SliderViewVerticalOne(model);

test('spyOn mouseDown toggle vertical one', () => {
  const somethingSpy = jest.spyOn(slider, 'onToggleMouseDown');
  let evt: any;
  slider.onToggleMouseDown(evt);
  expect(somethingSpy).toHaveBeenCalledTimes(1)
});

it('jest snapshots', () => {
  expect(slider.element).toMatchSnapshot()
})