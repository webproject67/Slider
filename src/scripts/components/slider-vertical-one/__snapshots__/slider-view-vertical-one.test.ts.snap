// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest snapshots 1`] = `
<div
  class="slider__wrapper"
>
  
      
  <div
    class="slider__inner slider__inner_with-height"
    data-testid="slider-vertical-one"
  >
    
        
    <div
      class="slider__scale slider__scale_with-height"
    >
      
          
      <div
        class="slider__bar slider__bar_with-width"
        style="height:100%"
      />
      
        
    </div>
    
        
    <div
      class="slider__toggle slider__toggle_vertical-maximum"
      style="top:100%"
    />
    
      
  </div>
  
    
</div>
`;
